cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/batch/sink/TimePartitionedFileSetDatasetParquetSink.java

```diff
@@ -0,0 +1,125 @@
+import co.cask.cdap.template.etl.api.PipelineConfigurer;
+import co.cask.cdap.template.etl.api.batch.BatchSink;
+import co.cask.cdap.template.etl.api.batch.BatchSinkContext;
+import co.cask.cdap.template.etl.common.StructuredToAvroTransformer;
+import com.google.common.collect.Maps;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.hadoop.mapreduce.Job;
+import parquet.avro.AvroParquetInputFormat;
+import parquet.avro.AvroParquetOutputFormat;
+
+import java.io.IOException;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * A {@link BatchSink} to write Parquet record to {@link TimePartitionedFileSet}
+ */
+@Plugin(type = "sink")
+@Name("TPFSParquet")
+@Description("PARQUET Sink with Time Partitioned File Dataset")
+public class TimePartitionedFileSetDatasetParquetSink extends
+  BatchSink<StructuredRecord, Void, GenericRecord> {
+
+  private static final String SCHEMA_DESC = "The schema of the record";
+  private static final String TPFS_NAME_DESC = "Name of the Time Partitioned FileSet Dataset to which the records " +
+    "have to be written";
+  private static final String BASE_PATH_DESC = "The base path for the time partitioned fileset. Defaults to the " +
+    "name of the dataset";
+  private final StructuredToAvroTransformer recordTransformer = new StructuredToAvroTransformer();
+
+  /**
+   * Config for TimePartitionedFileSetDatasetParquetSink
+   */
+  public static class TPFSParquetSinkConfig extends PluginConfig {
+
+    @Description(TPFS_NAME_DESC)
+    private String name;
+
+    @Description(SCHEMA_DESC)
+    private String schema;
+
+    @Description(BASE_PATH_DESC)
+    @Nullable
+    private String basePath;
+  }
+
+  private final TPFSParquetSinkConfig tpfsParquetSinkConfig;
+  private String hiveSchema;
+
+  public TimePartitionedFileSetDatasetParquetSink(TPFSParquetSinkConfig tpfsParquetSinkConfig)
+    throws IOException, UnsupportedTypeException {
+    this.tpfsParquetSinkConfig = tpfsParquetSinkConfig;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+
+    String tpfsName = tpfsParquetSinkConfig.name;
+    String basePath = tpfsParquetSinkConfig.basePath == null ? tpfsName : tpfsParquetSinkConfig.basePath;
+    try {
+      hiveSchema = SchemaConverter.toHiveSchema(
+        co.cask.cdap.api.data.schema.Schema.parseJson(tpfsParquetSinkConfig.schema.toLowerCase()));
+    } catch (UnsupportedTypeException e) {
+      //do nothing
+    } catch (IOException e) {
+      //do nothing
+    }
+    pipelineConfigurer.createDataset(tpfsName, TimePartitionedFileSet.class.getName(), FileSetProperties.builder()
+      .setBasePath(basePath)
+      .setInputFormat(AvroParquetInputFormat.class)
+      .setOutputFormat(AvroParquetOutputFormat.class)
+      .setEnableExploreOnCreate(true)
+      .setExploreFormat("PARQUET")
+      .setExploreSchema(hiveSchema.substring(1, hiveSchema.length() - 1))
```

Comment:
I think it's better to expand the FileSetProperties to accept `Schema` object to set the schema as well. @anew thoughts?